### PR TITLE
Filesystem Spinner on SessionEditFragment will update to newly created filesystem

### PR DESCRIPTION
**Pull request by luongthomas**
**Describe the pull request**

**Provide a description of the problem your pull request solves. Screenshots, code examples, etc are welcome in this section.**

In the session edit screen, after creating a new filesystem, the spinner will already have the newly created filesystem selected.

new_filesystem

**Link to relevant issues**
#164

### CODE REVIEW

**Comment by djangohacked** [comment]
I'm a beginner, I want to solve this problem please

**Comment by corbinlc** [comment]
There is already a pull request against it.  But, we encourage others
contributing.  Look at our project page, but you can probably also imagine
other things that could be improved.

**Comment by corbinlc** [comment]
@MatthewTighe can you review this?

**Comment by djangohacked** [comment]
Solution?

File: app/src/main/java/tech/ula/ui/SessionEditFragment.kt [hunk 1]
```diff
@@ -111,9 +117,10 @@ class SessionEditFragment : Fragment() {
                     else -> {
                         // TODO adapter to associate filesystem structure with list items?
                         val filesystem = filesystemList.find { it.name == filesystemName }
-                        session.filesystemName = filesystem!!.name
-                        session.filesystemId = filesystem.id
-                        text_input_username.setText(filesystem.defaultUsername)
+                        filesystem?.run {
```
Comment (MatthewTighe) [inline app/src/main/java/tech/ula/ui/SessionEditFragment.kt]: Will you change this to `filesysem?.let {` and `this` to `it` to remain consistent with how things have been done elsewhere in the code?

`run` is a slightly more specific use case than is needed here. 
[some standard library explanations](https://proandroiddev.com/the-tldr-on-kotlins-let-apply-also-with-and-run-functions-6253f06d152b)
[a stack overflow description of receivers](https://stackoverflow.com/questions/45875491/what-is-a-receiver-in-kotlin)

File: app/src/main/java/tech/ula/ui/SessionEditFragment.kt [hunk 2]
```diff
@@ -199,4 +206,16 @@ class SessionEditFragment : Fragment() {
...
     }
+
+    private fun getListDifferenceAndSetNewFilesystem(prevFilesystems: List<Filesystem>, currentFilesystems: List<Filesystem>) {
+        val uniqueFilesystems = currentFilesystems.subtract(prevFilesystems)
+        if (uniqueFilesystems.isNotEmpty()) {
+            setFilesystemForSessionTo(uniqueFilesystems.first())
+        }
+    }
+
+    private fun setFilesystemForSessionTo(filesystem: Filesystem) {
```
Comment (MatthewTighe) [inline app/src/main/java/tech/ula/ui/SessionEditFragment.kt]: I don't love the name of this one. Maybe something like `updateFilesystemDetailsForSession`?

File: app/src/main/java/tech/ula/ui/SessionEditFragment.kt (hunk 1 above)
Comment (luongthomas) [reply app/src/main/java/tech/ula/ui/SessionEditFragment.kt]: Yeah I can change it to let.  Thanks for providing some helpful links

**Pull request merged**